frontend/src/components/AppSearch.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+    <div class="search">
+        <input 
+            type='text' 
+            v-model='search' 
+            name='search' 
+            placeholder='Search...'
+        />
+        <ul v-if="search && suggestions.length">
+            <router-link 
+                :key="item.id"
+                tag="li" 
+                v-for="item in suggestions"
+                :to="{ path: `/course/${item.id}`}"
+            >
+                {{ item.name }} ({{ item.code }})
+            </router-link>
+        </ul>
+    </div>
+</template>
+
+
+<script>
+export default {
+    data() {
+        return {
+            search: '',
+            courses: []
```

Comment:
shouldn't this be in the store?

Reply:
courses? Maybe. Probably. 'search' needs to be here tho cos of v-model. But yeah I'll adapt soon.

Reply:
cool